jerry-core/parser/regexp/re-parser.cpp

```diff
@@ -365,18 +365,20 @@ re_parse_char_class (re_parser_ctx_t *parser_ctx_p, /**< number of classes */
       }
       else if (ch == 'd')
       {
-        /* append digits from '0' to '9'. */
-        append_char_class (re_ctx_p, 0x0030UL, 0x0039UL);
+        /* See ECMA-262 v5, 15.10.2.12 */
+        append_char_class (re_ctx_p, LIT_CHAR_ASCII_DIGITS_BEGIN, LIT_CHAR_ASCII_DIGITS_END);
         ch = RE_CHAR_UNDEF;
       }
       else if (ch == 'D')
       {
+        /* See ECMA-262 v5, 15.10.2.12 */
         append_char_class (re_ctx_p, 0x0000UL, 0x002FUL);
-        append_char_class (re_ctx_p, 0x003AUL, 0xFFFFUL);
+        append_char_class (re_ctx_p, 0x003AUL, LIT_UTF16_CODE_UNIT_MAX);
         ch = RE_CHAR_UNDEF;
       }
       else if (ch == 's')
       {
+        /* See ECMA-262 v5, 15.10.2.12 */
         append_char_class (re_ctx_p, 0x0009UL, 0x000DUL);
         append_char_class (re_ctx_p, 0x0020UL, 0x0020UL);
         append_char_class (re_ctx_p, 0x00A0UL, 0x00A0UL);
```

Comment:
@ruben-ayrapetyan,
"The production CharacterClassEscape :: s evaluates by returning the set of characters containing the characters that are on the right-hand side of the WhiteSpace (7.2) or LineTerminator (7.3) productions."